app/src/main/java/org/jellyfin/androidtv/util/apiclient/BaseItemUtils.kt

```diff
@@ -0,0 +1,224 @@
+import org.jellyfin.androidtv.util.Utils
+import org.jellyfin.apiclient.interaction.ApiClient
+import org.jellyfin.apiclient.model.dto.BaseItemDto
+import org.jellyfin.apiclient.model.dto.BaseItemType
+import org.jellyfin.apiclient.model.dto.ImageOptions
+import org.jellyfin.apiclient.model.entities.ImageType
+import org.jellyfin.apiclient.model.entities.LocationType
+import org.jellyfin.apiclient.model.entities.PersonType
+import org.jellyfin.apiclient.model.library.PlayAccess
+import org.jellyfin.apiclient.model.livetv.SeriesTimerInfoDto
+import org.koin.java.KoinJavaComponent.get
+import java.util.*
+
+// TODO Feature Envy!!! Wants to live in BaseItemDto.
+fun BaseItemDto.isLiveTv() =
+	baseItemType == BaseItemType.Program || baseItemType == BaseItemType.LiveTvChannel
+
+fun BaseItemDto.canPlay() = playAccess == PlayAccess.Full
+	&& ((isPlaceHolder == null || !isPlaceHolder)
+	&& (baseItemType != BaseItemType.Episode || locationType != LocationType.Virtual))
+	&& baseItemType != BaseItemType.Person
+	&& baseItemType != BaseItemType.SeriesTimer
+	&& (!isFolderItem || childCount == null || childCount > 0)
+
+fun BaseItemDto.getFullName(context: Context) = when (baseItemType) {
+	BaseItemType.Episode -> listOfNotNull(
+		seriesName,
+		when {
+			parentIndexNumber != null -> context.getString(R.string.lbl_season_number, parentIndexNumber)
+			else -> null
+		},
+		when {
+			indexNumber != null -> when {
+				indexNumberEnd != null -> context.getString(R.string.lbl_episode_range, indexNumber, indexNumberEnd)
+				else -> context.getString(R.string.lbl_episode_number, indexNumber)
+			}
+			else -> null
+		}
+	).filter { it.isNotEmpty() }.joinToString(" ")
+	// we actually want the artist name if available
+	BaseItemType.Audio,
+	BaseItemType.MusicAlbum -> listOfNotNull(albumArtist, name)
+		.filter { it.isNotEmpty() }
+		.joinToString(" - ")
+	else -> name.orEmpty()
+}
+
+fun BaseItemDto.getDisplayName(context: Context): String {
+	val seasonNumber = when {
+		baseItemType == BaseItemType.Episode
+			&& parentIndexNumber != null
+			&& parentIndexNumber != 0 ->
+			context.getString(R.string.lbl_season_number, parentIndexNumber)
+		else -> null
+	}
+	val episodeNumber = when {
+		baseItemType != BaseItemType.Episode -> indexNumber?.toString()
+		parentIndexNumber == 0 -> context.getString(R.string.lbl_special)
+		indexNumber != null -> when {
+			indexNumberEnd != null -> context.getString(R.string.lbl_episode_range, indexNumber, indexNumberEnd)
+			else -> context.getString(R.string.lbl_episode_number, indexNumber)
+		}
+		else -> null
+	}
+	val seasonEpisodeNumbers = listOfNotNull(seasonNumber, episodeNumber).joinToString(":")
+
+	val nameSeparator = when (baseItemType) {
+		BaseItemType.Episode -> " — "
+		else -> ". "
+	}
+
+	return listOfNotNull(seasonEpisodeNumbers, name)
+		.filter { it.isNotEmpty() }
+		.joinToString(nameSeparator)
+}
+
+fun BaseItemDto.getSubName(context: Context) = when (baseItemType) {
```

Comment:
Do we want the `.orEmtpy` calls in this function? I'd expect the usages of the function rather have a null value if there is no name.

Reply:
When I was originally testing this, the app crashed when it was loading the home screen because officialRating was null. I wasn’t sure if it was possible for something like that to happen if name were null too

Reply:
Weird, now it's not crashing.